### PR TITLE
store,tidb-server: add a morbid flag to indicate tidb server is closing

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

When tikv is closed first, and Ctrl+C to tidb then, the server may pending for a while.
Domain is waiting on ddl.Close(), and ddl is waiting for all its worker, the worker hang here:

```
goroutine 334 [select]:
github.com/pingcap/tidb/store/tikv.NewBackoffFn.func1(0x5660ec0, 0xc000afe5a0, 0x0)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/backoff.go:105 +0x3bd
github.com/pingcap/tidb/store/tikv.(*Backoffer).Backoff(0xc0009ba540, 0x0, 0x562e860, 0xc000867c60, 0x2, 0x562e860)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/backoff.go:275 +0x157
github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).onSendFail(0xc000993268, 0xc0009ba540, 0xc000863e80, 0x562e8a0, 0xc000867bc0, 0xc000506640, 0x4a817c800)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/region_request.go:172 +0x18f
github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).sendReqToRegion(0xc000993268, 0xc0009ba540, 0xc000863e80, 0xc000506640, 0x4a817c800, 0xc000863e80, 0x0, 0x0, 0x0)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/region_request.go:141 +0x10d
github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).SendReqCtx(0xc000993268, 0xc0009ba540, 0xc000506640, 0x4, 0x1, 0x2, 0x4a817c800, 0x0, 0x0, 0x0, ...)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/region_request.go:109 +0xde
github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).SendReq(...)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/region_request.go:65
github.com/pingcap/tidb/store/tikv.(*tikvSnapshot).get(0xc0009ec600, 0xc0009ba540, 0xc0008b4030, 0x1b, 0x23, 0xc000808398, 0x5660ec0, 0xc000afe5a0, 0xc0001ef040, 0xc0001ef040)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/snapshot.go:254 +0x204
github.com/pingcap/tidb/store/tikv.(*tikvSnapshot).Get(0xc0009ec600, 0xc0008b4030, 0x1b, 0x23, 0x0, 0x0, 0x0, 0x562ed00, 0xc0001ef040)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/snapshot.go:225 +0x12a
github.com/pingcap/tidb/kv.(*unionStore).Get(0xc0009ec660, 0xc0008b4030, 0x1b, 0x23, 0x66c1c00, 0x2, 0x6, 0x8, 0xc000ba7450)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/kv/union_store.go:192 +0x213
github.com/pingcap/tidb/store/tikv.(*tikvTxn).Get(0xc000b72fd0, 0xc0008b4030, 0x1b, 0x23, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/store/tikv/txn.go:130 +0xdb
github.com/pingcap/tidb/structure.(*TxStructure).loadListMeta(0xc000a6a400, 0xc0008b4030, 0x1b, 0x23, 0xc0008b4030, 0x1b, 0x23, 0xc000ba7548)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/structure/list.go:216 +0x5e
github.com/pingcap/tidb/structure.(*TxStructure).LIndex(0xc000a6a400, 0x637b7b0, 0xa, 0xa, 0x0, 0x2720c4e2ab40d, 0xc026e96d00, 0x26e96d0000b6b800, 0x5cdcf7f0, 0xc000ba75a0)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/structure/list.go:162 +0x79
github.com/pingcap/tidb/meta.(*Meta).getDDLJob(0xc000afe570, 0x637b7b0, 0xa, 0xa, 0x0, 0x490a153, 0x526ece0, 0xc000afe570)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/meta/meta.go:476 +0x69
github.com/pingcap/tidb/meta.(*Meta).GetDDLJobByIdx(0xc000afe570, 0x0, 0x0, 0x0, 0x0, 0xc000afe570, 0x24, 0x0)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/meta/meta.go:505 +0xd7
github.com/pingcap/tidb/ddl.(*worker).getFirstDDLJob(0xc0007654a0, 0xc000afe570, 0x53bfcfa, 0x7, 0xc000afe570)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/ddl/ddl_worker.go:222 +0x38
github.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue.func1(0x569c960, 0xc000b72fd0, 0xc000b72fd0, 0x0)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/ddl/ddl_worker.go:371 +0xda
github.com/pingcap/tidb/kv.RunInNewTxn(0x56804c0, 0xc00043a0f0, 0xc00004a900, 0xc000993d30, 0x3e, 0x8b)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/kv/txn.go:50 +0xde
github.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue(0xc0007654a0, 0xc000094d20, 0x24, 0xc000a6a300)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/ddl/ddl_worker.go:362 +0x13f
github.com/pingcap/tidb/ddl.(*worker).start(0xc0007654a0, 0xc000094d20)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/ddl/ddl_worker.go:134 +0x228
github.com/pingcap/tidb/ddl.(*ddl).start.func1()
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/ddl/ddl.go:442 +0x37
github.com/pingcap/tidb/util.WithRecovery(0xc000790240, 0xc000790260)
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/util/misc.go:81 +0x4d
created by github.com/pingcap/tidb/ddl.(*ddl).start
	/Users/breezewish/Work/PingCAP/src/github.com/pingcap/tidb/ddl/ddl.go:441 +0x559
```

### What is changed and how it works?

Because ddl worker waiting for response, and tikv client retrying the request, this situation lasts until timeout. 
tikv client needs retry to handle network error, but in this case we want it to exit ASAP
Add a morbid flag for that.

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: store/tikv/error.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ import (
 var (
 	// ErrBodyMissing response body is missing error
 	ErrBodyMissing = errors.New("response body is missing")
+	// When TiDB is closing and send request to tikv fail, do not retry, return this error.
+	errTiDBMoribid = errors.New("tidb server moribid")
```
Comment (XuHuaiyu) [inline store/tikv/error.go]: Will this error message be returned to the client?
If so, should we make it more readable for the users?

Comment (tiancaiamao) [reply store/tikv/error.go]: This state is set in `closeDomainAndStorage` which is called after `TryGracefulDown`:
https://github.com/pingcap/tidb/pull/10511/files#diff-af2c9279ed41f7023f9b6b73cba2d18fR582

After `TryGracefulDown`, connections to client have already been closed, so the error doesn't return to the client.

**Comment by tiancaiamao** [comment]
/run-all-tests
PTAL @XuHuaiyu

**Comment by morgo** [comment]
May I suggest renaming it from `Dying` to `ShuttingDown`?

Dying makes it sound like it is of poor health and maybe being evicted from a cluster (it is actually ambiguous, and not all users will know tidb is without cluster membership). ShuttingDown makes it clearer that this is intentional.

**Pull request merged**